src/test/java/org/mule/extension/email/value/MailboxFolderValueProviderTestCase.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.email.value;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.IsCollectionContaining.hasItems;
+import static org.mule.tck.junit4.matcher.ValueMatcher.valueWithId;
+
+import org.mule.extension.email.EmailConnectorTestCase;
+import org.mule.runtime.api.component.location.Location;
+import org.mule.runtime.api.value.ValueProviderService;
+import org.mule.runtime.api.value.ValueResult;
+import org.mule.tck.junit4.rule.SystemProperty;
+
+import javax.inject.Inject;
+
+import com.icegreen.greenmail.imap.ImapHostManager;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class MailboxFolderValueProviderTestCase extends EmailConnectorTestCase {
+
+  private static final String SPECIAL_CHARACTER_PASSWORD = "*uawH*IDXlh2p%21xSPOx%23%25zLpL";
+  private static final String SALESFORCE_FOLDER = "salesforce";
+  private static final String MULESOFT_FOLDER = SALESFORCE_FOLDER + "/mulesoft";
+
+  @Rule
+  public SystemProperty specialCharacterPassword = new SystemProperty("specialCharacterPassword", SPECIAL_CHARACTER_PASSWORD);
+
+  @Inject
+  private ValueProviderService service;
```

Comment:
To have a more authentic execution, a more similar to what happens in Studio, Value Provider/ Metadata tests should run on Lazy Mode that is done overriding some methods:

```java
  @Override
  public boolean enableLazyInit() {
    return true;
  }

  @Override
  public boolean disableXmlValidations() {
    return true;
  }
```

Also you will need to do some tunning on how to importe the `ValueProviderService`:

```java
@Inject
@Named(VALUE_PROVIDER_SERVICE_KEY)
private ValueProviderService service;
```